src/obj.rs

```diff
@@ -124,12 +125,15 @@ where
             normal: Vec::new(),
             objects: Vec::new(),
             material_libs: Vec::new(),
+            path: PathBuf::new(),
         }
     }
 
     pub fn load(path: &Path) -> io::Result<Obj<P>> {
         let f = File::open(path)?;
-        let obj = Obj::load_buf(&mut BufReader::new(f))?;
+        let mut obj = Obj::load_buf(&mut BufReader::new(f))?;
+        // unwrap is safe as we've read this file before
+        obj.path = path.parent().unwrap().to_owned();
```

Comment:
why does it need `parent()` here?

Reply:
Because the path is smth like "/long/path/file.obj", but we need only "/long/path/" part.